plotly/tools.py

```diff
@@ -1460,6 +1468,50 @@ def validate_streamline(x, y, **kwargs):
                                              "evenly spaced array")
 
     @staticmethod
+    def validate_ohlc(op, hi, lo, cl, **kwargs):
+        length = len(hi)
+        if any(len(lst) != length for lst in [op, lo, cl]):
+            raise exceptions.PlotlyError("Oops! Your high, open, low, and "
+                                         "close lists should all be the same "
+                                         "length.")
+
+        for lst in [op, lo, cl]:
+            for index in range(len(hi)):
+                if hi[index] < lst[index]:
+                    raise exceptions.PlotlyError("Oops! Looks like some of "
+                                                 "your high values are less "
+                                                 "the corresponding open, "
+                                                 "low, or close values. "
+                                                 "Double check that your data "
+                                                 "is entered in O-H-L-C order")
+
+        for lst in [hi, op, cl]:
+            for index in range(len(lo)):
+                if lo[index] > lst[index]:
+                    raise exceptions.PlotlyError("Oops! Looks like some of "
+                                                 "your low values are greater "
+                                                 "than the corresponding high"
+                                                 ", open, or close values. "
+                                                 "Double check that your data "
+                                                 "is entered in O-H-L-C order")
+
+    @staticmethod
+    def flatten(array):
+        """
+        Uses list comprehension to flatten array
+
+        :param (array): An iterable to flatten
+        :raises (PlotlyError): If iterable is not nested.
+        :rtype (list): The flattened list.
+        """
+        try:
+            return [item for sublist in array for item in sublist]
+        except TypeError:
+            raise exceptions.PlotlyError("Your data array could not be "
+                                         "flattened! Make sure x, y, u, and v "
+                                         "are lists or numpy ndarrays!")
```

Comment:
:cow2: this message could probably be improved. We need a `list or lists` or an `ndarray`? Is that right? Also, does it make sense to talk about `x, y, u, v`?